middle_end/flambda2.0/to_cmm/un_cps.ml

```diff
@@ -970,33 +981,35 @@ and wrap_call_exn env e call k_exn =
 
 (* Wrap a function call to honour its continuation *)
 and wrap_cont env effs res e =
-  let k = Apply_expr.continuation e in
-  if Continuation.equal (Env.return_cont env) k then
+  match Apply_expr.continuation e with
+  | Never_returns ->
+    res
```

Comment:
One possibility would be to generate a sequence of `res` followed by a `raise` operation. The `raise` will be discarded by the deadcode patch, and will allow `Selectgen` to correctly infer the branch doesn't return.

Reply:
Right, also, flambda would statically know that the branch raises an exception.

Reply:
There is now proper handling of never_returning extcalls in the backend (cmm and mach), so there is no need for the sequence.